opendevin/runtime/plugins/agent_skills/agentskills.py

```diff
@@ -20,12 +20,29 @@
 from inspect import signature
 from typing import Optional
 
+import base64
+import PyPDF2
+import docx
+from pptx import Presentation
+from pylatexenc.latex2text import LatexNodes2Text
+from openai import OpenAI
+
 CURRENT_FILE = None
 CURRENT_LINE = 1
 WINDOW = 100
 
 ENABLE_AUTO_LINT = os.getenv('ENABLE_AUTO_LINT', 'false').lower() == 'true'
 
+# OPENAI
+OPENAI_API_KEY = os.getenv('OPENAI_API_KEY', '')
```

Comment:
This will not work out of the box because those environmental variables are not available in the sandbox.

We need to rethink about the design here (including how to mock those API calls in the tests), but I think for now you could put those environmental variables into the sandbox in `__init__` function of `sandbox.py`

Reply:
I think these tools are primarily needed for GAIA (https://github.com/OpenDevin/OpenDevin/pull/1911) to perform well as proposed in https://github.com/OpenDevin/OpenDevin/pull/1914

In the default case, we should just disable these tools for the agent, and only enable them for evals now. In the case for GAIA evaluation, we can just set `SANDBOX_ OPENAI_API_KEY`.

I discussed with @rbren earlier, I think it will a more fundamental solution if we can connect these `agentskills` (executed inside sandbox) with `Event Stream` on our backend (e.g., every time a audio-to-text function is called by the agent inside sandbox, it push an `AudioToTextAction` to the `EventStream` on the backend - maybe by exposing a REST API from backend; The python function execution inside sandbox will be blocked until an `AudioToTextObservation` is processed by the backend and returned to the function in the sandbox).

Reply:
> In the case for GAIA evaluation, we can just set SANDBOX_OPENAI_API_KEY.

Okay yeah that makes sense for evaluation purposes

Reply:
Yep i'm actually hoping we can have a way to optionally switch on/off these tools - so that they won't get included into the prompt for normal tasks